client/client.go

```diff
@@ -396,6 +413,11 @@ func (client *Client) do(method, path string, query url.Values, headers map[stri
 	return rsp.StatusCode, nil
 }
 
+func notRetryable(err error) bool {
```

Comment:
For the sake of a little bikeshedding ;), I wonder if it would make sense to reverse the semantics here and follow the naming of the existing `ShouldRetryError(err) bool` from httputil, this would make it more easy to find all these helpers later.

Reply:
I opted not to do that because there were two options: 
a) have the function determine which errors are retryable instead of which are not retryable. If we do that and a new error is added later or I forget to include one now, then it's not retried (which is worse than unnecessarily retrying). 
b) write the function in the negative (like: `!(errors.Is(err, AuthorizationError{}) || errors.Is(err, InternalClientError{}))`) and then again reverse the result again when calling. It's safer but it seemed more convoluted than just changing the function name.
But I can change the naming to "shouldNotRetryError", to make it consistent with existing code